process/block/postprocess/basePostProcess.go

```diff
@@ -171,3 +179,54 @@ func (bpp *basePostProcessor) RemoveProcessedResultsFor(txHashes [][]byte) {
 		delete(bpp.mapTxToResult, string(txHash))
 	}
 }
+
+func (bpp *basePostProcessor) splitMiniBlocksIfNeeded(miniBlocks []*block.MiniBlock) []*block.MiniBlock {
+	splitMiniBlocks := make([]*block.MiniBlock, 0)
+	mapGasLimitInReceiverShard := make(map[uint32]uint64)
+
+	for _, miniBlock := range miniBlocks {
+		receiverShardID := miniBlock.ReceiverShardID
+		currentMiniBlock := createEmptyMiniBlock(miniBlock)
+		mapGasLimitInReceiverShard[receiverShardID] = 0
+
+		for _, txHash := range miniBlock.TxHashes {
```

Comment:
can we extract this inner loop into a separate function?

Reply:
done